src/Ontology/Oep4.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Oep4.h"
+#include <HexCoding.h>
+#include <cstdio>
+
+namespace TW::Ontology {
+Oep4::Oep4(const Address addr) noexcept
+    : oep4Contract(addr._data.begin(), addr._data.end()) {
```

Comment:
oep4Contract can probably take a `Data` in parameter instead and use `oep4Contract(addr._data)`

Reply:
Add another ctor